themes/default/content/docs/intro/concepts/resources/options/replaceonchanges.md

```diff
@@ -0,0 +1,100 @@
+
+{{% choosable language javascript %}}
+
+```javascript
+let widget = new k8s.apiextensions.CustomResource("widget", {
+    apiVersion: 'acmecorp.com/v1alpha1',
+    kind: 'Widget',
+    spec: {
+        input: "something",
+    },
+}, { replaceOnChanges: ["spec.input"] });
+```
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+let widget = new k8s.apiextensions.CustomResource("widget", {
+    apiVersion: 'acmecorp.com/v1alpha1',
+    kind: 'Widget',
+    spec: {
+        input: "something",
+    },
+}, { replaceOnChanges: ["spec.input"] });
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+widget = apiextensions.CustomResource("widget",
+    api_version="acmecorp.com/v1alpha1",
+    kind="Widget",
+    spec={
+        "input": "something",
+    },
+    opts=ResourceOptions(replace_on_changes=["spec.input"])
+)
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+widget, err = apiextensions.NewCustomResource(ctx, "widgt", &apiextensions.CustomResourceArgs{
+    ApiVersion: pulumi.String("acmecorp.com/v1alpha1"),
+    Kind:       pulumi.String("Widget"),
+    OtherFields: kubernetes.UntypedArgs{
+        "spec": map[string]interface{}{
+            "input": "something",
+        },
+    },
+}, pulumi.ReplaceOnChanges([]string{"spec.input"})
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var widget = new Pulumi.Kubernetes.ApiExtensions.CustomResource("widget", new WidgetArgs
+{
+    Spec = new WidgetSpecArgs
+    {
+        Input = "something",
+    }
+}, pulumi.ReplaceOnChanges([]string{"spec.input"}));
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+The property paths provided as input to `replaceOnChanges` can each describe a subset of the properties of the resource which should trigger a replacement on changes.  The `*` wildcard can be used in any part of a path.  A few examples of what changes will trigger replacement for a given property path string:
+
+- `*`: any property change
+- `spec`: any change to the `spec` property or any of its children
+- `spec[0]`: any change to the first element of the array in the `spec` property or any of its children
+- `spec[*].item`: any change to the `item` property of any element of the array in the `spec` prroperty or any of the `item` property's children
+
+{{% notes "info" %}}
+The property paths passed to `replaceOnChanges` should always be the "camelCase" version of the property name, as used in the core Pulumi resource model.
+{{% /notes %}}
+
+If there initialization errors on a resource (because the resource was created but failed to fully initialize correctly on a previous deployment), then this will typically cause the resource to be updated on the following Pulumi update, even if there are no other changes to the resources inputs.  If `*` is specified as a property path for `replaceOnChanges`, then initialization errors will trigger a replacement instead of an update.
```

Comment:
This sentence seems like it needs fixing, but I'm not exactly sure what it's trying to say. Maybe this?

```suggestion
If there are initialization errors on a resource (because the resource was created but failed to fully initialize correctly on a previous deployment), then this will typically cause the resource to be updated on the following Pulumi update, even if there are no other changes to the resource's inputs.  If `*` is specified as a property path for `replaceOnChanges`, then initialization errors will trigger a replacement instead of an update.
```